File: 0x08-python-more_classes/3-rectangle.py
#!/usr/bin/python3
""" Write a class Rectangle that defines a rectangle by:
(based on 0-rectangle.py)"""


class Rectangle():
    """pass"""
    def __init__(self, width=0, height=0):
        """the class Rectangle"""
        self.width = width
        self.height = height

    def area(self):
        """area of the Rectangle"""
        return self.height * self.width

    def perimeter(self):
        """ perimeter of the Rectangle"""
        if self.width == 0 or self.height == 0:
            return 0
        else:
            return self.height + self.height + self.width + self.width

    @property
    def width(self):
        """getter and setter"""
        return self.__width__

    @width.setter
    def width(self, value):
        """Set width"""
        if type(value) != int:
            raise TypeError("width must be an integer")
        elif value < 0:
            raise ValueError("width must be >= 0")
        self.__width__ = value

    @property
    def height(self):
        """The height"""
        return self.__height__

    @height.setter
    def height(self, value):
        """Setter an getter"""
        if type(value) != int:
            raise TypeError("height must be an integer")
        elif value < 0:
            raise ValueError("height must be >= 0")
        self.__height__ = value

    def __str__(self):
        """Class as a String"""
        strg = ''
        for i in range(self.height):
            strg += '#' * self.width + '\n'
        strg = strg.rstrip()
        return strg

File: 0x13-javascript_objects_scopes_closures/3-rectangle.js
#!/usr/bin/node
/*
Write a class Rectangle that defines a rectangle
If w or h is equal to 0 or not a positive integer,
create an empty object
Create an instance method called print()
that prints the rectangle using the character X
*/

class Rectangle {
  constructor (w, h) {
    if (w > 0 && h > 0) {
      this.width = w;
      this.height = h;
    }
  }

  print () {
    for (let i = 0; i < this.height; i++) {
      console.log('X'.repeat(this.width));
    }
  }
}
module.exports = Rectangle;
